### PR TITLE
Sodium fixes

### PR DESCRIPTION
**Pull request by jedisct1**
No changes to the resulting binary, but this simplifies possibly confusing code.

### CODE REVIEW

File: fizz/crypto/exchange/X25519.cpp [hunk 1]
```diff
@@ -46,17 +43,14 @@ std::unique_ptr<folly::IOBuf> X25519KeyExchange::generateSharedSecret(
...
   }
-  if (keyShare.size() != crypto_box_curve25519xsalsa20poly1305_PUBLICKEYBYTES) {
+  if (keyShare.size() != crypto_scalarmult_BYTES) {
     throw std::runtime_error("Invalid external public key");
   }
-  auto key = IOBuf::create(crypto_scalarmult_curve25519_BYTES);
+  auto key = IOBuf::create(crypto_scalarmult_BYTES);
   key->append(crypto_scalarmult_curve25519_BYTES);
-  int err = crypto_scalarmult_curve25519(
+  int err = crypto_scalarmult(
```
[the remark this answers is not in the document]
Comment (jedisct1) [reply fizz/crypto/exchange/X25519.cpp]: Yes. `crypto_scalarmult()` is guaranteed to always be X25519 unless there is a major revision of the library. This is valid for all high-level APIs.

File: fizz/crypto/exchange/X25519.cpp [hunk 2]
```diff
@@ -46,17 +43,14 @@ std::unique_ptr<folly::IOBuf> X25519KeyExchange::generateSharedSecret(
   if (!privKey_ || !pubKey_) {
     throw std::runtime_error("Key not generated");
   }
-  if (keyShare.size() != crypto_box_curve25519xsalsa20poly1305_PUBLICKEYBYTES) {
+  if (keyShare.size() != crypto_scalarmult_BYTES) {
     throw std::runtime_error("Invalid external public key");
   }
-  auto key = IOBuf::create(crypto_scalarmult_curve25519_BYTES);
+  auto key = IOBuf::create(crypto_scalarmult_BYTES);
   key->append(crypto_scalarmult_curve25519_BYTES);
```
[the remark this answers is not in the document]
Comment (jedisct1) [reply fizz/crypto/exchange/X25519.cpp]: Added, thanks!

File: fizz/crypto/exchange/X25519.cpp [hunk 3]
```diff
@@ -11,15 +11,12 @@
 #include <fizz/crypto/Utils.h>
 
 #include <folly/Conv.h>
-#include <sodium/crypto_scalarmult.h>
+#include <sodium.h>
 
 using namespace folly;
 
 namespace {
```
Comment (knekritz) [inline fizz/crypto/exchange/X25519.cpp]: The opening bracket is still here

File: fizz/crypto/exchange/X25519.cpp [hunk 4]
```diff
@@ -46,17 +41,14 @@ std::unique_ptr<folly::IOBuf> X25519KeyExchange::generateSharedSecret(
...
-  if (keyShare.size() != crypto_box_curve25519xsalsa20poly1305_PUBLICKEYBYTES) {
+  if (keyShare.size() != crypto_scalarmult_BYTES) {
     throw std::runtime_error("Invalid external public key");
   }
-  auto key = IOBuf::create(crypto_scalarmult_curve25519_BYTES);
-  key->append(crypto_scalarmult_curve25519_BYTES);
-  int err = crypto_scalarmult_curve25519(
+  auto key = IOBuf::create(crypto_scalarmult_BYTES);
+  key->append(crypto_scalarmult_BYTES);
+  int err = crypto_scalarmult(
```
Comment (siyengar) [inline fizz/crypto/exchange/X25519.cpp]: are you planning to keep the algorithm the same across libsodium versions?

Comment (siyengar) [reply fizz/crypto/exchange/X25519.cpp]: nvm i see you answered the same question for kyle.

**Comment by knekritz** [comment]
Merged, thanks!

**Pull request closed without merging**